osf/models/node.py

```diff
@@ -441,12 +441,39 @@ def is_preprint(self):
         if not self.preprint_file_id or not self.is_public:
             return False
         if self.preprint_file.node_id == self.id:
-            return self.has_published_preprint
+            return self.has_preprint_non_initial_state
         else:
             self._is_preprint_orphan = True
             return False
 
     @property
+    def has_preprint_non_initial_state(self):
+        return self.preprints.filter(~Q(reviews_state='initial')).exists()
+
+    @property
+    def has_moderated_preprint(self):
```

Comment:
These properties seem like they should be computed in the view, they're not useful on the model. Maybe a method here to get the preprint (prefer a published one, if it exists), then fill out the view data based on that?